spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindingService.java

```diff
@@ -128,6 +128,17 @@ public BindingService(
 		return binding;
 	}
 
+	@SuppressWarnings("rawtypes")
+	public Object getExtendedProducerProperties(Object output, String outputName) {
+		Binder binder =  getBinder(outputName, output.getClass());
+		if (binder instanceof ExtendedPropertiesBinder) {
+			return ((ExtendedPropertiesBinder) binder).getExtendedProducerProperties(outputName);
+		}
+		else {
+			return null;
+		}
```

Comment:
Should we just remove `else`?

Reply:
I generally prefer this style but don't really mind either way; the bytecode is identical.

Feel free to change on merge if you feel strongly.

Reply:
I am good